plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadataFactory.java

```diff
@@ -29,28 +29,29 @@ public class DefaultJdbcMetadataFactory
     private final JdbcClient jdbcClient;
     private final TimestampTimeZoneDomain timestampTimeZoneDomain;
     private final Set<JdbcQueryEventListener> jdbcQueryEventListeners;
+    private final IdentityCacheMapping identityCacheMapping;
 
     @Inject
     public DefaultJdbcMetadataFactory(
             JdbcClient jdbcClient,
             TimestampTimeZoneDomain timestampTimeZoneDomain,
-            Set<JdbcQueryEventListener> jdbcQueryEventListeners)
+            Set<JdbcQueryEventListener> jdbcQueryEventListeners,
+            IdentityCacheMapping identityCacheMapping)
     {
         this.jdbcClient = requireNonNull(jdbcClient, "jdbcClient is null");
         this.timestampTimeZoneDomain = requireNonNull(timestampTimeZoneDomain, "timestampTimeZoneDomain is null");
         this.jdbcQueryEventListeners = ImmutableSet.copyOf(requireNonNull(jdbcQueryEventListeners, "queryEventListeners is null"));
+        this.identityCacheMapping = requireNonNull(identityCacheMapping, "identityCacheMapping is null");
     }
 
     @Override
     public JdbcMetadata create(JdbcTransactionHandle transaction)
     {
-        // Session stays the same per transaction, therefore session properties don't need to
-        // be a part of cache keys in CachingJdbcClient.
```

Comment:
Why it isn't true anymore ?

Reply:
This is true for all cases, except those where we may want to modify session manually within transaction.